src/vs/platform/terminal/node/terminalProcess.ts

```diff
@@ -315,7 +317,6 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 			this._childProcessMonitor?.handleOutput();
 		});
 		ptyProcess.onExit(e => {
-			this._exitCode = e.exitCode;
```

Comment:
Likely the cause of the exit code/test problems

Reply:
woah yeah not sure how/why I deleted this 😅